### PR TITLE
removing InsecureSkipVerify

### LINKED ISSUE #341: remove `InsecureSkipVerify` when downloading images
see https://github.com/opiproject/sztp/blob/c70e8e35d37e079f84e8924075014b7b739e8092/sztp-agent/pkg/secureagent/daemon.go#L242

### PR DESCRIPTION
**Pull request by mohamedasifs123**
closes #341

### CODE REVIEW

**Comment by glimchb** [comment]
@mohamedasifs123 did you test it ? it fails ci for me...

**Review by glimchb — CHANGES_REQUESTED** [review]
@mohamedasifs123 did you test it ? it fails ci for me...

File: sztp-agent/pkg/secureagent/daemon.go [hunk 1]
```diff
@@ -225,9 +225,8 @@ func (a *Agent) downloadAndValidateImage() error {
 			Transport: &http.Transport{
 				TLSClientConfig: &tls.Config{
 					//nolint:gosec
```
Comment (bhoopesh369) [inline sztp-agent/pkg/secureagent/daemon.go:227]: this can be removed now

**Comment by mohamedasifs123** [comment]
Yeh for me too test falling. We need to come up with another way.

**Comment by glimchb** [comment]
> Yeh for me too test falling. We need to come up with another way.

so let's close this PR for now until we have a solutions

**Pull request closed without merging**